models/git/branch_list.go

```diff
@@ -59,30 +60,74 @@ func (branches BranchList) LoadPusher(ctx context.Context) error {
 	return nil
 }
 
+func (branches BranchList) LoadRepo(ctx context.Context) error {
+	ids := container.FilterSlice(branches, func(branch *Branch) (int64, bool) {
+		return branch.RepoID, branch.RepoID > 0 && branch.Repo == nil
+	})
+
+	reposMap := make(map[int64]*repo_model.Repository, len(ids))
+	if err := db.GetEngine(ctx).In("id", ids).Find(&reposMap); err != nil {
+		return err
+	}
+	for _, branch := range branches {
+		if branch.RepoID <= 0 || branch.Repo != nil {
+			continue
+		}
+		branch.Repo = reposMap[branch.RepoID]
+	}
+	return nil
+}
+
 type FindBranchOptions struct {
 	db.ListOptions
 	RepoID             int64
+	RepoCond           builder.Cond
 	ExcludeBranchNames []string
+	CommitCond         builder.Cond
+	PusherID           int64
```

Comment:
It doesn't seem good to continue bloating the `FindXxxOptions`. They are used for general querying purpose. Mixing too many options together would cause unclear behaviors and make it difficult to maintain.

Now since you have a clear requirement to query some "recently pushed branches", I think it needs a brand new function to do such query.

Reply:
Done in 484b3344f84f380777bcce5c007d602595434926